### ISSUE TITLE
Feature Request: Support for Custom Markdown Components

### ISSUE THREAD

**Issue by ikkysleepy** [imzbf/md-editor-v3#279]
Is it possible to support custom markdown component? 

I am trying to replace the image tag with a s3 image

```<s3-image key="path/to/my/image.jpg"/>```

I was able to upload a file , get the url  and insert the tag, but wanted to see I could insert my custom tag  for images?

**Comment by imzbf** [comment]
It's possible. 

We need to override `markdown-it` plugin [markdown-it-image-figures](https://github.com/Antonio-Laguna/markdown-it-image-figures/blob/main/src/index.js)

1. Copy source code from [markdown-it-image-figures](https://github.com/Antonio-Laguna/markdown-it-image-figures/blob/main/src/index.js)
2. Replace default tag with you custom tag after row 102
  ```diff
  102          image = token.children.length === 1 ? token.children[0] : token.children[1];
  + 103        image.tag = 's3-image';
  ```
3. Configure editor
  ```js
  import { config } from 'md-editor-v3';
  import ImageFiguresPlugin from 'markdown-it-image-figures.js';

  config({
    markdownItConfig: (mdit) => {
      mdit.use(ImageFiguresPlugin, { figcaption: true });
    },
  });
  ```

**Comment by imzbf** [comment]
There are some issues, the next patch version can be used like this.

**Comment by ikkysleepy** [comment]
Ok,  sounds good. Thanks for the quick response.

Is this how I would load the s3image vue component and modify the config:

```
    import s3image from '@/components/plugins/s3-image.vue';
    import { config } from 'md-editor-v3';
    import ImageFiguresPlugin from '@/composables/markdown-it-image-figures';

    config({
        markdownItConfig: (mdit) => {
            mdit.use(ImageFiguresPlugin, { figcaption: true });
        },
    });
    
    import { MdEditor, NormalToolbar } from 'md-editor-v3';
    import 'md-editor-v3/lib/style.css';
```

I added the line you talked about and copied the figure:

```
+ 103        image.tag = 's3image';
```

I'll wait for the next patch to see how this works. Thanks.

**Comment by ikkysleepy** [comment]
Thanks. I tested it out and it kind of works. I was trying to do an async call to get the base64 data from Amazon S3 to load and I did convert the function to async but even then it sometimes worked and sometimes did not. Mostly on the first image it worked but not the 2nd image.  I think things are just not setup for async calls. As an intermediate solution I am inserting the access_token of the user as a param to authenticate the user to return the decrypted image.  This did work even though images take a while to load, but they do load.  The only issue with this is that someone could share an image url w/the access token and be able to view the image until the token expires but not that big of a concern.
